extensions/ql-vscode/src/skeleton-query-wizard.ts

```diff
@@ -253,51 +265,51 @@ export class SkeletonQueryWizard {
     databaseNwo: string,
     databaseItems: readonly DatabaseItem[],
   ): Promise<DatabaseItem | undefined> {
-    const dbItems = databaseItems || [];
-    const dbs = dbItems.filter(
-      (db) =>
-        db.language === language &&
-        db.name === databaseNwo &&
-        db.error === undefined,
+    const dbs = databaseItems.filter(
+      (db) => db.language === language && db.name === databaseNwo,
     );
 
-    if (dbs.length === 0) {
-      return undefined;
-    }
-    return dbs[0];
+    return dbs.pop();
   }
 
   public async findDatabaseItemByLanguage(
     language: string,
     databaseItems: readonly DatabaseItem[],
   ): Promise<DatabaseItem | undefined> {
-    const dbItems = databaseItems || [];
-    const dbs = dbItems.filter(
-      (db) => db.language === language && db.error === undefined,
-    );
-    if (dbs.length === 0) {
-      return undefined;
-    }
-    return dbs[0];
+    const dbs = databaseItems.filter((db) => db.language === language);
+
+    return dbs.pop();
   }
 
-  private async findExistingDatabaseItem() {
-    if (this.language === undefined) {
-      throw new Error("Language is undefined");
-    }
+  public async findExistingDatabaseItem(
```

Comment:
These functions are a little confusing as to whether they should be members of the `SkeletonQueryWizard` class or not, because we're passing in arguments that are otherwise available on `this`. It makes the functions slightly harder to implement because you have to know whether there's a `language` argument available and that you should use that instead of `this.language`. We don't want to end up using both from the same method.

We could make these methods `static` so then that removes the chance of accidentally referencing `this`, or move them out of the class so they're just top-level non-exported methods.

Reply:
I've converted these types of methods into `static` versions.